docs/content/advanced/backend/events/events-architecture.md

```diff
@@ -0,0 +1,85 @@
+The events system in Medusa is built on a publish/subscribe architecture. Events are published by the core when certain actions take place. Those events can be subscribed to, which means, a reactive action is performed asynchronously. It is useful in a variety of scenarios including but not limited to processing heavy jobs in the background and reacting to specific events in a custom project or plugin. 
+
+### Publishing and subscribing
+
+The `EventBusService` is responsible for publishing and processing events.
+
+> The current implementation of the `EventBusService` is powered by Redis. We will soon introduce an event bus module, allowing you to use any pub/sub provider. Therefore, the following documentation is based on the Redis implementation. Still, the general purpose and flows of the `EventBusService` will remain the same with the module implementation.
+
+The `EventBusService` exposes two methods in its public API for event processing; `emit` and `subscribe`. The former will take your event, add it to a Bull queue (powered by Redis) as a job, and process it asynchronously. The latter will attach a listener/subscriber to the specified event, which is run upon processing it. In this document, the terms event and job are used interchangeably.
+
+The signature of each method and an example can be seen below:
+
+**Emit**
+```ts
+// signature
+async emit<T>(eventName: string, data: T, options: Record<string, unknown> & EmitOptions = { attempts: 1 }): Promise<StagedJob | void>
+
+// example
+eventBusService.emit("product.created", { id: "prod_id" }, { attempts: 2 })
```

Comment:
Suggestion: to be less pron to error should we suggest using ProducService.Events.CREATED instead of the string itself?

Reply:
As a general practice it seams to be better and less impacted by changes over time

Reply:
As the `emit` function can be used in a custom project, I actually believe the raw string is a better choice, as readers would otherwise get confused about the variables.

Reply:
I am fine with that, it is just that it is more prom to typo and if the value is changed in the core it will not be reflected for the user. It also reduce the intelisens